src/main/java/org/sagebionetworks/web/client/presenter/RejectReasonWidget.java

```diff
@@ -0,0 +1,151 @@
+package org.sagebionetworks.web.client.presenter;
+
+import com.google.gwt.user.client.rpc.AsyncCallback;
+import com.google.gwt.user.client.ui.IsWidget;
+import com.google.gwt.user.client.ui.Widget;
+import com.google.inject.Inject;
+import org.sagebionetworks.repo.model.UserProfile;
+import org.sagebionetworks.web.client.DisplayUtils;
+import org.sagebionetworks.web.client.utils.CallbackP;
+import org.sagebionetworks.web.client.widget.asynch.UserProfileAsyncHandler;
+import org.sagebionetworks.web.client.widget.entity.RejectReasonView;
+
+
+public class RejectReasonWidget implements RejectReasonView.Presenter, IsWidget {
+
+
+    // Template rejection format
+    public static String TEMPLATE_HEADER_HELLO = "Hi ";
+    public static String TEMPLATE_HEADER_THANKS = ",\n\nThank you for submitting your Synapse" +
+            " profile for validation. Before I can accept your request, you need to:\n";
+    public static String TEMPLATE_HEADER_SIGNATURE = "Let us know if you have any questions.\n\nRegards,\nact@sagebase.org";
+
+    // If no options are shown for rejected reason
+    public static String ERROR_MESSAGE = "Error: Please select at least one checkbox and generate a response or manually enter a response";
+
+    // Common reasons for user rejection
+    public static String REJECT_TAKE_SYNAPSE_QZ = "Take and pass the Synapse Certification quiz. ";
+    public static String REJECT_ADD_INFO = "Add at least one piece of information (education, employment, etc.) to your ORCID profile and set it to “public. ";
+    public static String REJECT_INITIAL_BOX = "Add at least one piece of information (education, employment, etc.) to your ORCID profile and set it to “public. ";
+    public static String REJECT_SUBMIT_DOCS = "Submit an accepted form of identity attestation documentation (e.g., letter from a signing official). ";
+    public static String REJECT_CUSTOM_REASON = "custom text to insert";
+
+    private String userName;
+    private RejectReasonView view;
+    UserProfileAsyncHandler handler;
+    CallbackP<String> callback;
+
+    @Inject
+    public RejectReasonWidget (UserProfileAsyncHandler handler, RejectReasonView view) {
+        this.handler = handler;
+        this.view = view;
+        this.view.setPresenter(this);
+        this.userName = "";
+        this.callback = null;
+    }
+
+    private void setUserProfileName(UserProfile user) {
+        this.userName = DisplayUtils.getDisplayName(user);
+    }
+
+    /*
+        For testing purposes only relay the userName to assert
+        that its being properly found.
+     */
+    public String getUserName() {
+        return this.userName;
+    }
+
+    // TODO: Remove when feature is in prod, used for demo purposes in ComingSoon
```

Comment:
This is the code review that will send these changes to prod.  You can remove from ComingSoon (and clean up this method).